includes/cli/class-amp-cli-validation-command.php

```diff
@@ -377,12 +400,237 @@ public function check_url( $args ) {
 			$url = $origin . '/' . ltrim( $url, '/' );
 		}
 
-		$result = AMP_Validation_Manager::validate_url( $url );
-		if ( $result instanceof WP_Error ) {
-			WP_CLI::error( $result );
+		$response = AMP_Validation_Manager::validate_url( $url );
+		if ( $response instanceof WP_Error ) {
+			WP_CLI::error( $response );
+		}
+
+		if ( 'count' === $assoc_args['format'] ) {
+			$count = count( $response['results'] );
+			WP_CLI::line( $count );
+			exit( $count ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		}
+
+		if ( empty( $response['results'] ) ) {
+			WP_CLI::success( 'The supplied URL does not contain any validation errors.' );
+			exit( 0 );
+		}
+
+		$errors = $this->flatten_validation_errors( $response['results'] );
+
+		if ( 'json' === $assoc_args['format'] ) {
+			WP_CLI::line( wp_json_encode( $errors, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES ) );
```

Comment:
If `json` I think it should output the entire `$response` here, as there is more data in the `$response` than just the resultant `$errors`.

```suggestion
			WP_CLI::line( wp_json_encode( $response, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES ) );
```

Reply:
As such, this `if ( 'json' === $assoc_args['format'] )` condition should be moved above the `if ( empty( $response['results'] ) ) ` check.